config/routes.rb

```diff
@@ -1,8 +1,11 @@
 # frozen_string_literal: true
 
 Rails.application.routes.draw do
-  get 'users', to: 'users#index'
-  get 'users/:user_id/posts', to: 'posts#index'
-  get 'users/:user_id/posts/:id', to: 'posts#show'
-  get 'users/:id', to: 'users#show'
+  root 'users#index'
+  resources :users,only:[:index, :show] do
+    resources :posts, only:[:index, :new, :create, :show] do
+    resources :comments, only: [:create]
```

Comment:
- [ ] Based on the requirements of this project, they should be a pair of routes for creating comments, that's 

```ruby
POST   /posts/:post_id/comments(.:format)     comments#create
 GET   /posts/:post_id/comments/new(.:format)     comments#new
```

Note that you are missing the new action in your comments controllers and you should equally have a view for it.

Reply:
Thanks @FrederickMih, let me do it right away!